x-pack/plugins/apm/server/routes/environments/get_environments.ts

```diff
@@ -39,44 +39,29 @@ export async function getEnvironments({
     ? 'get_environments_for_service'
     : 'get_environments';
 
-  const params = {
+  const params: APMEventESTermsEnumRequest = {
     apm: {
       events: [
         getProcessorEventForTransactions(searchAggregatedTransactions),
         ProcessorEvent.metric,
         ProcessorEvent.error,
       ],
     },
-    body: {
-      track_total_hits: false,
-      size: 0,
-      query: {
-        bool: {
-          filter: [
-            ...rangeQuery(start, end),
-            ...termQuery(SERVICE_NAME, serviceName),
-          ],
-        },
-      },
-      aggs: {
-        environments: {
-          terms: {
-            field: SERVICE_ENVIRONMENT,
-            missing: ENVIRONMENT_NOT_DEFINED.value,
-            size,
-          },
-        },
+    size,
+    field: SERVICE_ENVIRONMENT,
+    index_filter: {
+      bool: {
+        filter: [...rangeQuery(start, end)],
       },
     },
   };
 
-  const resp = await apmEventClient.search(operationName, params);
-  const aggs = resp.aggregations;
-  const environmentsBuckets = aggs?.environments.buckets || [];
+  if (serviceName) {
+    params.string = serviceName;
+  }
```

Comment:
Do you really need this here? Can't you do something like:

Since serviceName is optional, initialize it
```
serviceName = ''
```

Then it looks like there's no problem on passing an empty string it keeps fetching all values.
```js
const params = {
 ...,
string: serviceName
}
```

Reply:
I was thinking that when serviceName is not passed, that is when Service Inventory page is loaded, then string will be passed as `''` which i wanted to avoid. But now after testing it looks like there will be no change in the API response, so i will add it back 👍🏼